File: Code Chef/C Programming/800-899/831 Mileage matters MILEAGE.c
#include <stdio.h>
int main()
{
    int t,n;
    float x,y,a,b;
    scanf("%d",&t);
    for(int i=0;i<t;i++)
    {
        scanf("%d %f %f %f %f",&n,&x,&y,&a,&b);
        float pc=((n/a)*x);
        float dc=((n/b)*y);
        if(pc==dc)
        {
            printf("ANY\n");
        }
        else if(pc<dc)
        {
            printf("PETROL\n");
        }
        else if(pc>dc)
        {
            printf("DIESEL\n");
        }
    }
    return 0;
}

File: Code Chef/JavaScript/800-899/831 Mileage matters MILEAGE.js
let input = '';
process.stdin.on('data', (data) => input += data);

const strToInt = (strArray) => {
    strArray.shift();
    const intArray = strArray.map((strItem) => strItem.trim().split(' ').map(Number));
    
    return intArray;
};

const getCarWithLessTravelCost = (distance, petrolCost, dieselCost, petrolMileage, dieselMileage) => {
  const petrolCarTravelCost = (distance * petrolCost) / petrolMileage;
  const dieselCarTravelCost = (distance * dieselCost) / dieselMileage;
  
  const carToRent = (petrolCarTravelCost < dieselCarTravelCost) ? 'PETROL'
    : (dieselCarTravelCost < petrolCarTravelCost) ? 'DIESEL'
    : 'ANY'
  ;
  console.log(carToRent);
};

process.stdin.on('end', () => {
    const testcases = strToInt(input.trim().split('\n'));
    
    testcases.forEach((testcase) => {
        const [distance, petrolCost, dieselCost, petrolMileage, dieselMileage] = testcase;
        getCarWithLessTravelCost(distance, petrolCost, dieselCost, petrolMileage, dieselMileage);
    });
});
